### PR TITLE
6918 integrate analysis api

### LINKED ISSUE #6918: Integrate analysis API
Integrate the cartodb.js analysis API

- [x] create analysis when new definition is created
- [x] remove analysis when definition is removed
- [x] analyze on initial reset
- [x] update (affected) analysis on definition change

### PR DESCRIPTION
**Pull request by viddo**
Resolves #6918 

@javierarce please review
cc @alonsogarciapablo @xavijam

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/data/analysis-definitions-collection.js [hunk 1]
```diff
@@ -65,10 +77,26 @@ module.exports = Backbone.Collection.extend({
...
+  _onRemove: function (m) {
+    var id = m.get('node_id');
+    var node = this._analysis.findNodeById(id);
+    if (node) {
+      node.remove();
+    }
+  },
+
+  _onAnalysisDefinitionNodeChange: function (m) {
+    this._analysis.analyse(m.toJSON());
```
Comment (viddo) [inline lib/assets/javascripts/cartodb3/data/analysis-definitions-collection.js]: @alonsogarciapablo this was done before your [commit this weekend](https://github.com/CartoDB/cartodb.js/commit/8ef5edc3d0a1ad92c0672878a6216f60c89850ae#commitcomment-16773257), as far as I understood the API should handle this correctly internally?

File: package.json [hunk 2]
```diff
@@ -27,9 +27,9 @@
     "browserify-shim": "3.8.10",
     "camshaft-reference": "^0.1.0",
     "cartoassets": "cartodb/CartoAssets#master",
-    "cartodb-deep-insights.js": "cartodb/deep-insights.js#master",
+    "cartodb-deep-insights.js": "cartodb/deep-insights.js#analysis",
     "cartodb-pecan": "0.2.x",
-    "cartodb.js": "cartodb/cartodb.js#v4",
+    "cartodb.js": "cartodb/cartodb.js#analysis",
```
Comment (viddo) [inline package.json]: Both the changes here are required until `analysis` branch is merged (to `v4`), requires that special setup locally for things to work too for the time being.

Comment (xavijam) [inline package.json]: :+1:, but the sooner we have the branch in v4 the bette.

File: lib/assets/javascripts/cartodb3/data/analysis-definitions-collection.js (hunk 1 above)
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb3/data/analysis-definitions-collection.js]: That's right!

**Comment by javierarce** [comment]
:+1:

**Comment by viddo** [comment]
will be merged into `analysis` as new "master" branch for the analysis WIP, until the dependencies are stable & available in production.

**Pull request closed without merging**